ndk-build/src/manifest.rs

```diff
@@ -258,6 +262,37 @@ pub struct Permission {
     pub max_sdk_version: Option<u32>,
 }
 
+/// Android [package element](https://developer.android.com/guide/topics/manifest/queries-element#package).
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct Package {
+    #[serde(rename(serialize = "android:name"))]
+    pub name: String,
+}
+
+/// Android [provider element](https://developer.android.com/guide/topics/manifest/queries-element#provider).
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct QueryProvider {
+    #[serde(rename(serialize = "android:authorities"))]
+    pub authorities: String,
+
+    // The specs say only an `authorities` attribute is required for providers contained in a `queries` element
+    // however aapt throws errors requiring an `android:name` attribute,
+    // This is possibly due to ndk-build using aapt instead of aapt2?
```

Comment:
I think we should validate this instead of merging in this question, should be pretty trivial to find out?

Reply:
Is there some other way you can validate this? It seems a bit of an odd "TODO" to have lingering in here. IIRC just creating an empty APK with the serialized manifest from here should work. Cobbling together an `aapt2` invocation from past experiences:

```console
$ aapt2 link -o test.apk --manifest 
"target/debug/apk/examples/AndroidManifest.xml" -I "~/Android/Sdk/platforms/android-32/android.jar" 
```

Reply:
I tested this for you. `cargo apk` with its `aapt` invocation indeed fails with:

```console
$ ~/Android/Sdk/build-tools/33.0.0-rc1/aapt package -f -F ./android-ndk-rs/target/debug/apk/examples/hello_world-unaligned.apk -M AndroidManifest.xml -I ~/Android/Sdk/platforms/android-29/android.jar
AndroidManifest.xml:1: Tag <provider> missing required attribute name.
```

Building with `aapt2` succeeds with a warning:

```console
$ aapt2 link -o test.apk --manifest target/debug/apk/examples/AndroidManifest.xml -I ~/Android/Sdk/platforms/android-32/android.jar
aapt2 W 04-16 12:44:51 212267 212267 LoadedArsc.cpp:657] Unknown chunk type '200'.
```

Subsequently reading `AndroidManifest.xml` from the successfully created APK shows that indeed only an `android:authorities` attribute shows up:

```console
aapt2 dump xmltree --file AndroidManifest.xml test.apk
N: android=http://schemas.android.com/apk/res/android (line=1)
  E: manifest (line=1)
...
      E: queries (line=1)
...
          E: provider (line=1)
            A: http://schemas.android.com/apk/res/android:authorities(0x01010018)="org.khronos.openxr.runtime_broker;org.khronos.openxr.system_runtime_broker" (Raw: "org.khronos.openxr.runtime_broker;org.khronos.openxr.system_runtime_broker")
...
```

If you can, please update the documentation and comments to note that this is required for `aapt` support and should be dropped if/when `cargo-apk` migrates to `aapt2`.